### PR TITLE
feat(core): proposal pre-analysis stage 2 — conflict analyzer (closes #204)

### LINKED ISSUE #204: Spec 55 §7.3: Proposal pre-analysis stage 2 — conflict analyzer
## Problem

PR #190 implemented Spec 55 §7.3 stages 1 (deduplication) and 3 (impact), with stage 2 (conflict detection) stubbed via a typed placeholder `ConflictPreAnalyzerStub`.

A Proposal-vs-Proposal / Proposal-vs-Rule / Proposal-vs-State conflict analyzer is still missing.

## Scope

- Analyzer module: `packages/core/src/life-system/proposal-preanalysis/conflict-analyzer.ts`
- Detect contradictions between:
  - The pending Proposal and other queued Proposals (e.g. two pending Rules with overlapping conditions but conflicting effects)
  - The pending Proposal and live Rule definitions (would cause unreachable rule logic)
  - The pending Proposal and live State definitions (would invalidate existing state transitions)
- Replace the stub registration in the default pipeline composition

## Refs

- Spec 55 `docs/specs/55_evolution_system.md` §7.3
- Stub: `packages/core/src/life-system/proposal-preanalysis/types.ts` (`ConflictPreAnalyzerStub`)
- Parent PR: #190
- Parent issue: #166

### PR DESCRIPTION
**Pull request by laofahai**
## Summary
- Replaces `ConflictPreAnalyzerStub` with a working stage-2 analyzer (Spec 55 §7.3).
- Detects three conflict shapes:
  - proposal-vs-proposal (shared `(target, name)` artifact, both with non-empty payloads)
  - proposal-vs-rule (proposed rule name collides with live rule)
  - proposal-vs-state (state-definition update/delete that orphans an existing transition)
- Optional `liveRules` / `liveStates` stores; analyzer skips probes whose store is omitted and records that in `notes`.
- Source-level failures captured into `notes` rather than thrown — one bad registry never nukes the pipeline.
- Both sync and async list functions accepted (Promise<T[]> | T[]).

## Test plan
- [x] `bun test packages/core/src/life-system/proposal-preanalysis/` — 39/39 (15 new)
- [x] `bun test` — 4573 pass / 0 fail
- [x] `bun run typecheck` — clean
- [x] `bun run check` — clean

## Refs
- Spec 55 §7.3 (`docs/specs/55_evolution_system.md`)
- Parent PR: #190
- Parent issue: #166

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Proposals now include conflict detection against existing rules and state definitions.
  * Added robust validation to identify proposal overlaps with pending submissions.

* **Tests**
  * Comprehensive test suite for conflict detection scenarios, including edge cases and error handling.

### CODE REVIEW

File: packages/core/src/life-system/proposal-preanalysis/conflict-analyzer.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            const proposedStateSet = new Set(proposed.states);
+            const removed: string[] = [];
+            for (const liveState of existing.states) {
+              if (!proposedStateSet.has(liveState)) removed.push(liveState);
+            }
+            if (removed.length === 0) continue;
+
+            for (const transition of existing.transitions) {
+              const refs = statesReferencedByTransition(transition);
+              const dropped = refs.filter((r) => removed.includes(r));
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/life-system/proposal-preanalysis/conflict-analyzer.ts]: Fixed in 3a7f0dd: `removed` is now a `Set<string>`, transition loop uses `removed.has()` for O(1) lookup. Complexity drops from O(T·R) to O(T+R).

File: packages/core/src/life-system/proposal-preanalysis/conflict-analyzer.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  );
+}
+
+/** Try to read a `StateDefinition`-shaped object out of an unknown change definition. */
+function readStateDefinition(value: unknown): StateDefinition | null {
+  if (!value || typeof value !== "object") return null;
+  const def = value as Partial<StateDefinition>;
+  if (!Array.isArray(def.states)) return null;
+  if (typeof def.name !== "string") return null;
+  return def as StateDefinition;
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/life-system/proposal-preanalysis/conflict-analyzer.ts:115]: Fixed in 3a7f0dd: `readStateDefinition` now requires every `states` entry to be a string; non-string entries cause the definition to be treated as malformed and surface a `notes` entry instead of triggering false transition conflicts. Added a regression test covering this.
